api/filtering/db_custom_filters.py

```diff
@@ -136,22 +136,18 @@ def separate_operating_system_filters(filter_url_params) -> list[OsFilter]:
     for filter_key in filter_url_params.keys():
         if filter_key == "name":
             ((os_comparator, os_name),) = filter_url_params[filter_key].items()
-            check_valid_os_name(os_name)
             version_node = {os_comparator: [None]}
         else:
             os_name = filter_key
-            check_valid_os_name(os_name)
-            if not isinstance(version_node := filter_url_params[os_name]["version"], dict):
+            if not isinstance(version_node := filter_url_params[os_name].get("version"), dict):
                 # If there's no comparator, treat it as "eq"
                 version_node = {"eq": version_node}
 
-        for os_comparator in version_node.keys():
-            version_array = version_node[os_comparator]
-            if not isinstance(version_array, list):
-                version_array = [version_array]
-
-            for version in version_array:
-                os_filter_list.append(OsFilter(os_name, os_comparator, version))
+        if isinstance(os_name, list):
+            for name in os_name:
+                os_filter_list += create_os_filter(name, version_node, os_filter_list)
+        else:
+            os_filter_list = create_os_filter(os_name, version_node, os_filter_list)
```

Comment:
Just to simplify the flow a bit (continued in create_os_filter)
```suggestion
        if not isinstance(os_name, list):
            os_name = [os_name]
        for name in os_name:
            os_filter_list += create_os_filter(name, version_node)
```